app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt

```diff
@@ -1,27 +1,130 @@
+import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
+const val DEFAULT_PAGE_SIZE = 5
+const val DEFAULT_NUMBER_OF_PAGE = 1
+const val DEFAULT_CURRENT_PAGE = 1
+
 @HiltViewModel
-class HomeViewModel @Inject constructor() : ViewModel() {
+class HomeViewModel @Inject constructor(
+    private val homeRepo: HomeRepo,
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
+) : ViewModel() {
 
-    private val _isRefreshing = MutableStateFlow(false)
     private var _error = MutableStateFlow<ErrorModel?>(null)
-    private var _surveyList = MutableStateFlow<ArrayList<SurveyResponse>>(arrayListOf())
+    private val _isRefreshing = MutableStateFlow(false)
+    private val _surveyList = MutableStateFlow<List<Survey>>(listOf())
+    private val _currentPageNumber = MutableStateFlow(DEFAULT_CURRENT_PAGE)
+    private val _userAvatar = MutableStateFlow<String?>(null)
+    private var totalNumberOfPage = DEFAULT_NUMBER_OF_PAGE
 
-    val error: StateFlow<ErrorModel?>
-        get() = _error.asStateFlow()
+    init {
+        getUserDetail()
+        getSurveyListFromDb()
+        getSurveyList()
+    }
 
-    val surveyList: StateFlow<ArrayList<SurveyResponse>>
+    val surveyList: StateFlow<List<Survey>>
         get() = _surveyList.asStateFlow()
 
     val isRefreshing: StateFlow<Boolean>
         get() = _isRefreshing.asStateFlow()
+
+    val error: StateFlow<ErrorModel?>
+        get() = _error.asStateFlow()
+
+    val userAvatar: StateFlow<String?>
+        get() = _userAvatar.asStateFlow()
+
+    private fun getUserDetail() {
+        viewModelScope.launch(ioDispatcher) {
+            homeRepo.fetchUserDetail().collect { result ->
+                when (result) {
+                    is ResourceState.Success -> {
+                        _userAvatar.value = result.data.attributes?.avatarUrl
+                        resetError()
+                    }
+                    else -> {
+                        result.mapError()?.let { errorModel ->
+                            _error.value = errorModel
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fun getNextPage() {
+        if (!_isRefreshing.value) {
+            _currentPageNumber.value += 1
+            getSurveyList()
+        }
+    }
+
+    fun clearCacheAndFetch() {
+        viewModelScope.launch(ioDispatcher) {
+            homeRepo.clearSurveyList()
+            _currentPageNumber.value = DEFAULT_CURRENT_PAGE
+            totalNumberOfPage = DEFAULT_NUMBER_OF_PAGE
+            getSurveyListFromDb()
+            getSurveyList()
+        }
+    }
+
+    private fun getSurveyListFromDb() {
+        viewModelScope.launch(ioDispatcher) {
+            homeRepo.getSurveyListFromDb().collect {
+                _surveyList.value = it
```

Comment:
Surely we should expose a domain-level `Survey` object, not the database entity directly? 🤔

Reply:
@ryan-conway Added and exposed `SurveyModel` 👍